samples/kotlin-native/build.gradle.kts

```diff
@@ -8,14 +8,14 @@ apply {
     plugin("org.jetbrains.kotlin.native")
 }
 
-kotlinNativeSourceSets["main"].component {
+kotlinNativeSourceSets["main"].component(Action {
     // https://github.com/JetBrains/kotlin-native/issues/1807
     if (this is KotlinNativeMainComponent) {
         outputKinds.set(listOf(OutputKind.EXECUTABLE))
         baseName.set("foo")
         extraOpts(listOf("-entry", "org.jlleitschuh.gradle.ktlint.sample.native.main"))
     }
-}
+})
```

Comment:
Is this `Action` required now?? Gradle 4.10 shouldn't make this necessary.

Reply:
Gradle actually complained, as it can't detect second param type, so I has to use `Action {}`

Reply:
Can you open an issue here:
https://github.com/gradle/kotlin-dsl/issues

Once done, leave a comment above this mentioning the issue link explaining `Action` is required.

I'll approve this PR after that.

Reply:
I don't think this is DSL problem, but rather native plugin itself:
```kotlin
e: /home/egorr/data/work/ktlint-gradle/samples/kotlin-native/build.gradle.kts:11:32: Overload resolution ambiguity: 
public open fun component(configureAction: Action<in AbstractKotlinNativeComponent>): KotlinNativeSourceSetImpl defined in org.jetbrains.kotlin.gradle.plugin.experimental.sourcesets.KotlinNativeSourceSetImpl
public open fun component(configureLambda: AbstractKotlinNativeComponent.() -> Unit): KotlinNativeSourceSetImpl defined in org.jetbrains.kotlin.gradle.plugin.experimental.sourcesets.KotlinNativeSourceSetImpl
```

Reply:
Yea, it's a bug that the Gradle Kotlin DSL should be aware of and resolve correctly.

Reply:
See: https://github.com/gradle/kotlin-dsl/issues/1138